src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+  //
+  for (unsigned int ix = 0; ix < eVec.size(); ++ix)
+  {
+    BSONObj entity = eVec[ix].embeddedObject();
+
+    if (!entity.hasField("id"))
+    {
+      LM_W(("Runtime Error (got a subscription without id)"));
+      continue;
+    }
+
+    std::string id = entity.getStringField("id");
+    
+    if (!entity.hasField("isPattern"))
+    {
+      continue;
+    }
+
+    std::string isPattern = entity.getStringField("isPattern");
+    if (isPattern != "true")
+    {
+      continue;
+    }
+
+    std::string  type = "";
+    if (entity.hasField("type"))
+    {
+      type = entity.getStringField("type");
+    }
+
+    EntityInfo* eiP = new EntityInfo(id, type);
+    eiV.push_back(eiP);
+  }
+
+  if (eiV.size() == 0)
+  {
+    return;
+  }
+
+
+  //
+  // 03. Pushing attribute names to Attribute Vector (attrV)
+  //
+  for (unsigned int ix = 0; ix < attrVec.size(); ++ix)
+  {
+    std::string attributeName = attrVec[ix].String();
+
+    attrV.push_back(attributeName);
+  }
+
+
+  //
+  // 04. FIXME P4: Restriction not implemented
+  //
+  
+  
+
+  //
+  // 05. Fill in notifyConditionVector from condVec
+  //
+  for (unsigned int ix = 0; ix < condVec.size(); ++ix)
+  {
+    BSONObj                   condition = condVec[ix].embeddedObject();
+    std::string               condType;
+    std::vector<BSONElement>  valueVec;
+
+    condType = condition.getStringField("type");
+    if (condType != "ONCHANGE")
+    {
+      continue;
+    }
+
+    NotifyCondition* ncP = new NotifyCondition();
+    ncP->type = condType;
+
+    valueVec = condition.getField("value").Array();
+    for (unsigned int vIx = 0; vIx < valueVec.size(); ++vIx)
+    {
+      std::string condValue;
+
+      condValue = valueVec[vIx].String();
+      ncP->condValueList.push_back(condValue);
+    }
+
+    notifyConditionVector.push_back(ncP);
+  }
+
+
+
+  //
+  // 06. Create Subscription and add it to the subscription-cache
+  //
+  Subscription* subP = new Subscription(tenant,
+                                        servicePath,
+                                        subId,
+                                        eiV,
+                                        attrV,
+                                        throttling,
+                                        expiration,
+                                        restriction,
+                                        notifyConditionVector,
+                                        reference,
+                                        format);
+  subCache->insert(subP);
+}
+
+
+
+/* ****************************************************************************
+*
+* SubscriptionCache::init - 
+*/
+void SubscriptionCache::init(void)
+{
+  semInit();
+  fillFromDb();
+}
+
+
+
+/* ****************************************************************************
+*
+* SubscriptionCache::fillFromDb - 
+*/
+void SubscriptionCache::fillFromDb(void)
+{
+  // FIXME P10: Get a list of all tenants and call subscriptionsTreat() for each of them
```

Comment:
The issue raised by this FIXME is important. If not implemented, I understand the cache doesn't work with ´-multiservice`.